tests/unittests/ExpressionTests.cpp

```diff
@@ -1239,4 +1239,69 @@ endmodule
     REQUIRE(diags.size() == 2);
     CHECK(diags[0].code == diag::BadRangeExpression);
     CHECK(diags[1].code == diag::BadRangeExpression);
-}
\ No newline at end of file
+}
+
+std::string testStringLiteralsToByteArray(const std::string& text) {
+    const auto& fullText = "module Top; " + text + " endmodule";
+    auto tree = SyntaxTree::fromText(string_view(fullText));
+
+    Compilation compilation;
+    compilation.addSyntaxTree(tree);
+    NO_COMPILATION_ERRORS;
+
+    const auto& module = *compilation.getRoot().topInstances[0];
+    if (!tree->diagnostics().empty())
+        WARN(report(tree->diagnostics()));
+
+    const ParameterSymbol& param = module.body.memberAt<ParameterSymbol>(0);
+    const auto& value = param.getValue();
+    std::string result;
+    if (value.isUnpacked()) {
+        for (const auto& svInt : value.elements()) {
+            REQUIRE(svInt.isInteger());
```

Comment:
This is fine, but .integer() will already assert so you don't actually have to REQUIRE() isInteger here. Doesn't hurt though.

Reply:
I remove the redundant code as suggested.